SConstruct

```diff
@@ -367,6 +387,35 @@ for d in os.listdir('.'):
         if os.path.exists(script):
             SConscript(script, variant_dir=join(GetOption('build-dir'), d), duplicate=0)
 
+if enscons_active:
+    Import('python_shlib')
+    Import('python_source')
+    Import('soarlib')
+    py_lib_namespace = env['PACKAGE_METADATA']['name'].replace("-", "_")
+
+    sources = []
+
+    if sys.platform == 'darwin' or os.name == 'nt':
+        # Add soar's library to the wheel directory
+        sources += [
+            env.Install(py_lib_namespace, soarlib)
+        ]
```

Comment:
MacOS and Windows require Soar's library to be distributed.

For some reason, the macos delocate (dynamic library fixer) does not fetch libSoar.dylib from the `out/` directory and statically link it, so this makes it happy.

Windows does not have a (stable) dynamic library fixer, so we need to just ship the dynamic library with it.

Reply:
Not sure I follow. Linux doesn't require the Soar library to be distributed? How do you run Soar there, then?

Reply:
Soar gets statically linked into the final extension module `.so` file, this is what [`auditwheel`](https://github.com/pypa/auditwheel) does :)

> `auditwheel repair`: copies these external shared libraries into the wheel itself, and automatically modifies the appropriate `RPATH` entries such that these libraries will be picked up at runtime. This accomplishes a similar result as if the libraries had been statically linked without requiring changes to the build system. Packagers are advised that bundling, like static linking, may implicate copyright concerns.

This effectively makes sure that python *can* load the extension module, regardless of wether the shared library exists in the final target, and this is how this extension essentially re-distributes Soar.

The MacOS `delocate` should do something similar, but for whatever reason, it is happy with the `.dylib` just sitting in the same folder, which effectively accomplishes the same. I couldn't get it to reliably pick up on the shared library sitting elsewhere, which is where the `DYLD_*` messing around comes from, so when I just packaged it with the wheel, it was happy, and worked all around.

For Windows, there exists no (stable) similar command yet, so we need to manually copy it in there.

Reply:
Thank you, this is very useful information! Could you add some comments to this effect?

Reply:
Done!